src/sdam/description/server.rs

```diff
@@ -106,7 +107,36 @@ pub(crate) struct ServerDescription {
     // allows us to ensure that only valid states are possible (e.g. preventing that both an error
     // and a reply are present) while still making it easy to define helper methods on
     // ServerDescription for information we need from the hello reply by propagating with `?`.
-    pub(crate) reply: Result<Option<HelloReply>, String>,
+    pub(crate) reply: Result<Option<HelloReply>>,
+}
+
+impl Serialize for ServerDescription {
```

Comment:
This manual implementation here just replicates what the old generated one was before the error change.